pkg/gofr/logging/logger_test.go

```diff
@@ -3,13 +3,16 @@ package logging
 import (
```

Comment:
Commenting a random line

Reply:
Here is the current logger implementation

```go
type logger struct {
        level      Level
        normalOut  io.Writer
        errorOut   io.Writer
        isTerminal bool
        lock       chan struct{}
}

// … redacted

func (l *logger) Fatal(args ...interface{}) {
        l.logf(FATAL, "", args...)

        //nolint:revive // exit status is 1 as it denotes failure as signified by Fatal log
        os.Exit(1)
}

func (l *logger) Fatalf(format string, args ...interface{}) {
        l.logf(FATAL, format, args...)

        //nolint:revive // exit status is 1 as it denotes failure as signified by Fatal log
        os.Exit(1)
}
```

Reply:
Here is what I'm suggesting to consider

```go
type logger struct {
        level      Level
        normalOut  io.Writer
        errorOut   io.Writer
        isTerminal bool
        lock       chan struct{}
        exiter func()
}

// … redacted

func (l *logger) Fatal(args ...interface{}) {
        l.logf(FATAL, "", args...)

         l.exitFatal()
}

func (l *logger) Fatalf(format string, args ...interface{}) {
        l.logf(FATAL, format, args...)

        l.exitFatal()
}

func (l *logger) exitFatal() {
        if l.exiter != nil {
            // this is used for test purpose
            l.exiter()
            return
        }

        //nolint:revive // exit status is 1 as it denotes failure as signified by Fatal log
        os.Exit(1)
}

func SetExiterT(t *testing.T, l *logger, f func()) {
     t.Helper()

     l.exiter = f
}
```

Then in tests, using

```go
var called bool
logging.SetExiter(t, logger, func() {
     called = true
})

Then check for called value

Reply:
I think just for the testing purpose, we should not change the struct and actual implementation.. here for testing purpose we are creating a new function and toggling on if condition for actual and testing case. IMO, this should not be done.

Reply:
I know

What about using what xhd2015 suggested here then?

https://github.com/gofr-dev/gofr/issues/844#issuecomment-2230608103

Reply:
But I'm fine with current implementation, even if it's a dirty hack for me